adapters/rubicon/rubicon.go

```diff
@@ -231,6 +233,15 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 }
 
 func (a *RubiconAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	requestCopy := request
+
+	err := openrtb_ext.ConvertUpTo26(&openrtb_ext.RequestWrapper{BidRequest: requestCopy})
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	request = requestCopy
+
```

Comment:
`MakeRequest` should not modify original request. This is a part of JSON test verification. It's easy to accidentally modify a property (or sub-object) in a shallow copy of an object such as any pointer in a bid request: `Site *Site`,  `App *App`, `Device *Device`, `User *User`, etc.

This code is what is causing the race condition error.
 
`ConvertUpTo26` sets `User.Eids` to original `request` and `requestCopy` because `user` is the same object in memory in both of them.

To fix this issue `user` should be copied from the original request and set back to `requestCopy` so `requestCopy` has its own `user`.  

Insert this code after line 236:

```go
if request.User != nil {
		userCopy := *request.User
		requestCopy.User = &userCopy
	}
```
This fixes race condition error